TaskSharper.Calender.WPF/ViewModels/CalendarWeekViewModel.cs

```diff
@@ -25,67 +25,52 @@ public class CalendarWeekViewModel : BindableBase
         private IEventAggregator _eventAggregator;
 
         public DelegateCommand NextCommand { get; set; }
-
         public DelegateCommand PrevCommand { get; set; }
 
         public ObservableCollection<CalendarDateViewModel> DateHeaders { get; set; }
         public ObservableCollection<CalendarEventsViewModel> EventContainers { get; set; }
 
         
-        public int WeeklyOffset { get; set; } = 0;
+        public DateTime CurrentWeek { get; set; }
 
 
         public CalendarWeekViewModel(ICalendarService service, IEventAggregator eventAggregator)
         {
-
-
-            NextCommand = new DelegateCommand(NextWeek);
-            PrevCommand = new DelegateCommand(PrevWeek);
             _service = service;
             _eventAggregator = eventAggregator;
 
+            NextCommand = new DelegateCommand(NextWeek);
+            PrevCommand = new DelegateCommand(PrevWeek);
+            
             DateHeaders = new ObservableCollection<CalendarDateViewModel>();
             EventContainers = new ObservableCollection<CalendarEventsViewModel>();
 
-            _eventAggregator.GetEvent<SpinnerEvent>().Publish(EventResources.SpinnerEnum.Show);
+            CurrentWeek = DateTime.Now;
 
             InitializeViews();
-            Task.Run(GetCalendarEvents);
-            
-            _eventAggregator.GetEvent<SpinnerEvent>().Publish(EventResources.SpinnerEnum.Hide);
         }
 
+        #region Commands
         private void NextWeek()
         {
-            WeeklyOffset++;
-            ReconstructView();
+            CurrentWeek = CurrentWeek.Date.AddDays(7);
+            _eventAggregator.GetEvent<WeekChangedEvent>().Publish(WeekChangedEnum.Increase);
         }
 
         private void PrevWeek()
```

Comment:
Use full "PreviousWeek" instead of "Prev"

Reply:
Agree. Done